sqlitecluster/SQLiteNode.cpp

```diff
@@ -1779,14 +1926,44 @@ void SQLiteNode::_changeState(SQLiteNode::State newState) {
     // Exclusively lock the stateMutex, nobody else will be able to get a shared lock until this is released.
     unique_lock<decltype(stateMutex)> lock(stateMutex);
 
-    // We send any unsent transactions here before we finish switching states. Normally, this does nothing, unless
-    // we're switching down from LEADING or STANDINGDOWN, but we need to make sure these are all sent to the new
-    // leader before we complete the transition.
-    _sendOutstandingTransactions();
-
     // Did we actually change _state?
     State oldState = _state;
     if (newState != oldState) {
+        // If we were following, and now we're not, we give up an any replications.
+        if (oldState == FOLLOWING) {
+            {
+                SINFO("Stopping following locking");
+                unique_lock<mutex> lock(_replicationCommandMutex);
+                SINFO("Stopping following locked");
+                // Clear the replication queue.
+                while (true) {
+                    try {
+                        _replicationCommands.pop();
+                    } catch (const out_of_range& e) {
```

Comment:
I'd suggest adding a `SSynchronizedQueue::clear()` command (which just calls `list::clear()`) to make this more obvious and performant.